Adafruit_MQTT.cpp

```diff
@@ -578,7 +578,7 @@ uint8_t Adafruit_MQTT::connectPacket(uint8_t *packet) {
   p[0] = MQTT_CONN_CLEANSESSION;
 
   // set the will flags if needed
-  if (will_topic && pgm_read_byte(will_topic) != 0) {
+  if (will_topic && will_topic[0] != '\0') {
```

Comment:
@eutill  Why are you comparing `\0` (null char.) here instead of `0` (int) from the previous will set?

Reply:
Because I didn't know that comparing to `0` (int) was equivalent for checking for null-termination.

Please feel free to commit on this.